com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/linking/LazyLinkingResource2.java

```diff
@@ -184,7 +188,22 @@ public synchronized EObject getEObject(final String uriFragment) {
    */
   @Override
   public <T> T getService(final Class<? extends T> key) {
-    return injector.getInstance(key);
+    IResourceSetServiceCache serviceCache = ServiceCacheAdapter.getServiceCacheAdapter(getResourceSet());
+    String fileExtension = getURI().fileExtension();
+    Map<Class<?>, Object> map = serviceCache.get(fileExtension);
+    if (map == null) {
+      map = new HashMap<>();
+      serviceCache.put(fileExtension, map);
+    }
+    @SuppressWarnings("unchecked")
+    T clazz = (T) map.get(key);

```

Comment:
'clazz' is usually used as the name for variables of type Class<>, but here the variable is of type T. Maybe 'service' would be a better name?

Reply:
You are right, I have changed it.